lib/cc/engine/analyzers/helpers/main.rb

```diff
@@ -0,0 +1,73 @@
+require 'pathname'
+
+module CC
+  module Engine
+    module Analyzers
+      module Helpers
+        def flay
+          @flay ||= ::Flay.new(options)
+        end
+
+        def new_violation(issue, location, other)
+          {
+            "type": "issue",
+            "check_name": name(issue),
+            "description": "Duplication found in #{issue.name}",
+            "categories": ["Duplication"],
+            "location": format_location(location),
+            "other_locations": format_locations(other)
+          }
+        end
+
+        def name(issue)
+          issue.identical? ? 'Identical code' : 'Similar code'
+        end
+
+        def find_other_locations(all_locations, current)
+          all_locations.reject { |location| location == current }
+        end
+
+        def format_location(location)
+          {
+            "path": local_path(location.file),
+            "lines": {
+              "begin": location.line,
+              "end": location.line
+            }
+          }
+        end
+
+        def format_locations(other)
+          other.map { |location| format_location(location) }
+        end
+
+        def options
```

Comment:
nit, but since you use this module as an include, I could see a day when somebody includes it in a runner-type class that implements it's own `options` method, and confusion ensues. `flay_options` might be clearer?